plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -649,6 +650,82 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
+     *     <li>git fetch --depth=1 origin scmTag</li>
+     *     <li>git checkout FETCH_HEAD</li>
+     * </ul>
+     * @param connectionURL can have a format such scm:git:https://github.com/jenkinsci/mailer-plugin.git or https://github.com/jenkinsci/mailer-plugin.git
+     * @param scmTag the tag or sha1 to clone
+     * @param checkoutDirectory the directory where to clone the git repository
+     * @param shallowClone if <code>true</code> the fetch will use <code>--depth=1</code>
+     * @return success or not
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public static boolean clone(String connectionURL, String scmTag, File checkoutDirectory, boolean shallowClone)
+            throws IOException, InterruptedException {
+
+        // maven scm is doing all of this, not sure why...
+        // 'git' 'clone' '--depth' '1' '--branch' 'scmTag' 'git url'
+        // 'git' 'ls-remote' 'giturl'
+        // 'git' 'fetch' 'giturl'
+        // 'git' 'checkout' 'scmTag'
+        // 'git' 'ls-files'
+
+        // we try something quick as the maven scm is doing clone and another useless fetch
+        // git init
+        // git remote add origin url
+        // git fetch --depth=1 origin scmTag (this will worl with SHA1 or tag)
+        // git checkout FETCH_HEAD
+
+        if (Files.exists(checkoutDirectory.toPath())) {
+            org.apache.commons.io.FileUtils.deleteDirectory(checkoutDirectory);
+        }
+        if (!checkoutDirectory.mkdirs()) {
+            System.out.println("cannot create directory " + checkoutDirectory);
+            return false;
+        }
+
+        // TODO timeout??
+        int res = new ProcessBuilder().directory(checkoutDirectory).command("git", "init").inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git init failed");
+            return false;
+        }
+        String gitUrl = connectionURL;
+        if (StringUtils.startsWith(connectionURL, "scm:git:")) {
+            gitUrl = StringUtils.substringAfter(connectionURL, "scm:git:");
+        }
+        res = new ProcessBuilder().directory(checkoutDirectory).command("git", "remote", "add", "origin", gitUrl).inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git remote add failed");
+            return false;
+        }
+        List<String> commands = new ArrayList<>();
+        commands.add("git");
+        commands.add("fetch");
+        if (shallowClone) {
+            commands.add("--depth=1");
+        }
+        commands.add("origin");
+        commands.add(scmTag);
+        res = new ProcessBuilder().directory(checkoutDirectory).command(commands).inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git fetch origin failed");
+            return false;
+        }
+        res = new ProcessBuilder().directory(checkoutDirectory).command("git", "checkout", "FETCH_HEAD").inheritIO().start().waitFor();
```

Comment:
Seems like too much work.

```bash
git clone --branch=$tag --depth=1 $url $dir
```

should suffice.

Reply:
nope as explained in the comment here https://github.com/jenkinsci/plugin-compat-tester/pull/382/files#diff-56ea8384acb1f763e2f462f02781e29ee845b4901b11f30bef2bd83ee984d51eR682
we need a solution working for both real tag and sha1.
some poms doesn't contain a real tag in their pom in the section
```
  <scm>
    ...
    <tag></tag>
    ...
  </scm>
```
such here https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/workflow/workflow-cps-global-lib/609.vd95673f149b_b/workflow-cps-global-lib-609.vd95673f149b_b.pom

Reply:
Hmm, I thought `--branch` worked with a commit hash but maybe I was imagining it?

Anyway you can at least skip the `git remote add`, as `git fetch` can accept a URL directly.